hal/common/file.hpp

```diff
@@ -65,7 +65,7 @@ class File
 	 * 
 	 * \return  success
 	 */
-	virtual bool open(const char* path) = 0;
+	virtual bool open(const char* path, bool new_file) = 0;
```

Comment:
No, I want to force the use of a new file in data_logging as in flash, you want to reuse the same file over and over again. 
As you are using the same class (in Linux), you need a parameter to tell whether you want to reuse the same file or create a new one.